experimental/dds/tree2/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -173,6 +187,36 @@ function composeMarks<TNodeChange>(
 		composeChild,
 	);
 
+	if (markIsTransient(newMark)) {
+		return withNodeChange(baseMark, nodeChange);
+	}
+	if (markIsTransient(baseMark)) {
+		if (isGenerativeMark(newMark)) {
+			// TODO: Make `withNodeChange` preserve type information so we don't need to cast here
+			const nonTransient = withNodeChange(
+				baseMark,
+				nodeChange,
+			) as GenerativeMark<TNodeChange>;
```

Comment:
This is safe because `markIsTransient` implies the mark is a `GenerativeMark`, right?

Reply:
Yes.